three.js/src/markers-area/arjs-markersareacontrols.js

```diff
@@ -36,7 +42,7 @@ const MarkersAreaControls = function (arToolkitContext, object3d, parameters) {
 };
 
 MarkersAreaControls.prototype = Object.create(ArBaseControls.prototype);
-MarkersAreaControls.prototype.constructor = MarkersAreaControls;
+//MarkersAreaControls.prototype.constructor = MarkersAreaControls;
```

Comment:
@nickw1  this is odd. I think this is an error in the code design. Actually the constructor of the class is created at line 6:
```js
const MarkersAreaControls = function (arToolkitContext, object3d, parameters) {
  var _this = this;
  ArBaseControls.call(this, object3d);
```
 I have to comment out this line because `tsc` compiler will create another constructor as a property function instead with a resulting error.

Reply:
@kalwalt I'm not familiar with this code, but I would strongly support using ES6 classes over older methods.

Reply:
I think that AR.js can be improved a lot in terms of readability if rewritten using ES6 classes. Not sure if this task it can be done before or after adding type definitions. This PR is a kind of test at the moment.

Reply:
Anyway removing that line has no side effect, i recompiled the libs and tested most of the examples with success. I think this was an old bug.

Reply:
Reading in the mdn documentation [prototype.constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/constructor) is actually the constructor of the created object (class). So not a mistake but maybe it's not needed. Anyway i will continue to convert the code ot Typescript and a more modern standard.